### PR TITLE
 Replaced the AlliedMissiles boolean on JamsMissiles by a Stances field

### PR DESCRIPTION
**Pull request by abc013**
Fixes the first part of #9382. (`JamsRadar` will be updated in another PR.)

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/JamsMissiles.cs [hunk 1]
```diff
@@ -13,10 +13,16 @@
 
 namespace OpenRA.Mods.Common.Traits
 {
+	[Desc("This actor deflects missiles.")]
```
Comment (Phrohdoh) [inline OpenRA.Mods.Common/Traits/JamsMissiles.cs]: This doesn't deflect so much as it does disable/destroy them.

Comment (pchote) [inline OpenRA.Mods.Common/Traits/JamsMissiles.cs]: It actually does deflect them – see the Missile code.

**Comment by abcdefg30** [comment]
Needs a rebase.

**Comment by abc013** [comment]
Rebased.

**Comment by GraionDilach** [comment]
:+1:

**Comment by chrisforbes** [comment]
:+1: LGTM.

**Comment by abcdefg30** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/e420e451d9ce8dfd14cd838116440994c569cff0...d18ff7dfb81225f4d546a1d4dbee192487c26a13)

**Pull request merged**